lexer.html

```diff
@@ -0,0 +1,140 @@
+<link rel="import" href="common.html">
+<script>
+function ShadeyCssLexer() {
+  'use strict';
+  // NOTE(cdata): Generally, tokens are structured as three-valued tuple,
+  // where the first value is the token value, the second value is the token
+  // type and the third value is the cursor index after consuming the raw
+  // source.
+
+  var common = ShadeyCssCommon();
+  var matcher = common.matcher;
+  var tokenType = common.tokenType;
+
+  function Whitespace(source, offset) {
+    var start = offset;
+
+    while (matcher.whitespace.test(source[offset])) {
+      ++offset;
+    }
+
+    return [
+      source.substr(start, offset - start),
+      tokenType.whitespace,
+      offset
+    ];
+  }
+
+  function Boundary(source, offset) {
+    var token;
+
+    if (matcher.boundary.test(source[offset])) {
+      return [
+        source[offset],
+        tokenType.boundary,
+        offset + 1
+      ];
+    }
+
+    return [
+      null,
+      tokenType.none,
+      offset
+    ];
+  }
+
+  function StringValue(source, offset) {
+    var quotation = source[offset];
+    var escaped = false;
+    var string = quotation;
+    var character;
+
+    if (!matcher.quotation.test(quotation)) {
+      throw new Error('Expected quotation mark, but got ' + quotation + ' at ' + offset);
+    }
+
+    while (character = source[++offset]) {
+      string += character;
+
+      if (escaped) {
+        escaped = false;
+        continue;
+      }
+
+      if (character === quotation) {
+        return [
+          string,
+          tokenType.string,
+          offset + 1
+        ];
+      }
+
+      if (character === '\\') {
+        escaped = true;
+      }
+    }
+
+    return [
+      null,
+      tokenType.none,
+      offset
+    ];
+  }
+
+  function Word(source, offset) {
+    var word = null;
+    var character;
+
+    while ((character = source[offset]) && !matcher.boundary.test(character)) {
+      word = word || '';
+      word += character;
+      ++offset;
+    }
+
+    return [
+      word,
+      word == null ? tokenType.none : tokenType.word,
+      offset
+    ];
+  }
+
+  function Token(source, offset) {
+    var token = Whitespace(source, offset);
+
+    if (token[0]) {
+      return token;
+    }
+
+    token = Boundary(source, token[2]);
```

Comment:
If you eliminate the whitespace and the boundaries, what is left are words and strings. But, I will re-implement so that this logic is more clear, and move it to a function with a better name.